extensions/cms/src/cmsResource/tree/registeredServerTreeNode.ts

```diff
@@ -38,7 +38,7 @@ export class RegisteredServerTreeNode extends CmsResourceTreeNodeBase {
 			serverName: this.serverName,
 			databaseName: '',
 			userName: undefined,
-			password: undefined,
+			password: '',
```

Comment:
Why?

Reply:
It caused errors in the admin-ext-tools extension because it assumes a connection always has a password.

Reply:
should the fix be in admin-ext-tools?  In general I don't like empty string to stand in for undefined, though it looks like we're doing that in databaseName and groupFullName already.

Reply:
Where in the extension are you seeing that it requires a password?

---
In reply to: [299743551](https://github.com/microsoft/azuredatastudio/pull/6242#discussion_r299743551) [](ancestors = 299743551)

Reply:
In the main file, it tries to retrieve the password from credential service

Reply:
Which it's still going to do even if the password is empty. I'm not seeing how this change would affect anything. 

```
	if (!password || password === '') {
		const creds = await azdata.connection.getCredentials(connectionContext.connectionProfile.id);
		password = creds[azdata.ConnectionOptionSpecialType.password];
	}
```